keps/0019-package-api-versioning.md

```diff
@@ -2,64 +2,370 @@
+## Proposals
+
+### With regards to package version and operator version
+
+Two proposals: singular and composite package/operator version. In the case of potential operator "bundles" (operators containing multiple app versions) both would end up "hiding" those app versions behind the package/operator version.
+
+#### Singular package/operator version (largely, the current state of things)
+
+##### Pros
+
+- Visually simple
+- It can be known just from the package version whether or not it contains a certain operator-related feature or bug fix
+
+##### Cons
+
+- Operator users care about the application version, not a "synthetic" version
+- Makes it complex to have an "app version"-based backport strategy. Customers might not be able to use operator based on newer app versions due to constraints outside of our control, but would still want to receive new operator-related features and bug fixes
+- Continuing from the second "pro", it might be possible that subsequent package versions with different app versions don't contain an operator-related feature or bug fix that only makes sense for a specific app version (e.g., version N+1 based on app version M doesn't include an operator feature available on version N based on app version M+1 because the operator feature only makes sense for app version M)
+- The progression of the package version is confusing with regards to the application version progression
+
+  Example:
+
+  | Time | Package/operator version | App version        | Comment                                            |
+  | ---- | ---------------          | ------------------ | -------------------------------------------------- |
+  | T0   | 1.0.0                    | 2.3.0              | Initial release based on app version 2.3.x         |
+  | T1   | 2.0.0                    | 3.0.0              | Release based on app version 3.0.x                 |
+  | T2   | 2.1.0                    | 2.3.0              | New operator-related feature for app version 2.3.x |
+  | T2   | 2.2.0                    | 3.0.0              | New operator-related feature for app version 3.0.x |
+  | T3   | 2.3.0                    | 2.3.1              | Bug fix release for app version 2.3.x              |
+
+  The example above illustrates two sources of confusion:
+  - A higher package version can contain a lower app version
+  - There's no relation between the package version growth and the app version growth
+
+#### Composite package/operator version
+
+##### Pros
+
+- Puts the app version (what the user cares about) front and center
+- Makes it trivial to have an "app version"-based backport/release strategy
+- Makes it trivial to have an "app version"-based support strategy (e.g., an organization communicates official support for only the N-2 versions of an operator)
+
+##### Cons
+
+- Visually more complex
+- Makes it complex to know which operator-related features are available in which versions
```

Comment:
+ operator version provides no information.  It's same  as having a 4 digit version `2.3.0.1` 2.3.0.2`  where `2.3.0` is app version and last digit is operator release

Reply:
We stated that we don't control the app versions e.g. `2.3.0+build.1848` is [100% valid SemVer](https://regex101.com/r/Ly7O1x/3/). And the only reason we would take _another_ SemVer for the operator version was to signal breaking changes.

Reply:
but operator-revision still doesn't provide any relevant information. `0.1.0` means different things based on which app version it's going. 
`2.3.0-0.1.0` doesn't have necessarily the same features as `3.0.0-0.1.0`